Client/Frontend/Browser/URLBarView.swift

```diff
@@ -326,87 +426,17 @@ class URLBarView: UIView, BrowserLocationViewDelegate, UITextFieldDelegate {
     }
 }
 
-/* Code for drawing the urlbar curve */
-// Curve's aspect ratio
-private let ASPECT_RATIO = 0.729
-
-// Width multipliers
-private let W_M1 = 0.343
-private let W_M2 = 0.514
-private let W_M3 = 0.49
-private let W_M4 = 0.545
-private let W_M5 = 0.723
-
-// Height multipliers
-private let H_M1 = 0.25
-private let H_M2 = 0.5
-private let H_M3 = 0.72
-private let H_M4 = 0.961
-
-private class CurveView: UIView {
-
-    func getWidthForHeight(height: Double) -> Double {
-        return height * ASPECT_RATIO
-    }
-
-    func drawFromTop(path: UIBezierPath) {
-        let height: Double = Double(AppConstants.ToolbarHeight)
-        let width = getWidthForHeight(height)
-        let from = (Double(self.frame.width) - width * 2 - Double(URLBarViewUX.URLBarCurveOffset - URLBarViewUX.URLBarCurveBounceBuffer), Double(AppConstants.StatusBarHeight))
-
-        path.moveToPoint(CGPoint(x: from.0, y: from.1))
-        path.addCurveToPoint(CGPoint(x: from.0 + width * W_M2, y: from.1 + height * H_M2),
-              controlPoint1: CGPoint(x: from.0 + width * W_M1, y: from.1),
-              controlPoint2: CGPoint(x: from.0 + width * W_M3, y: from.1 + height * H_M1))
-
-        path.addCurveToPoint(CGPoint(x: from.0 + width,        y: from.1 + height),
-              controlPoint1: CGPoint(x: from.0 + width * W_M4, y: from.1 + height * H_M3),
-              controlPoint2: CGPoint(x: from.0 + width * W_M5, y: from.1 + height * H_M4))
-    }
-
-    private func getPath() -> UIBezierPath {
-        let path = UIBezierPath()
-        self.drawFromTop(path)
-        path.addLineToPoint(CGPoint(x: self.frame.width, y: AppConstants.ToolbarHeight + AppConstants.StatusBarHeight))
-        path.addLineToPoint(CGPoint(x: self.frame.width, y: 0))
-        path.addLineToPoint(CGPoint(x: 0, y: 0))
-        path.addLineToPoint(CGPoint(x: 0, y: AppConstants.StatusBarHeight))
-        path.closePath()
-        return path
-    }
-
-    override class func layerClass() -> AnyClass {
-        return CAShapeLayer.self
+extension URLBarView: BrowserToolbarProtocol {
```

Comment:
It might make more sense to include this protocol at the `URLBarView` class definition in this case. All of the button getters are define there, so there's some disconnect between those and the methods down here (and I missed this extension on the first pass).